platform/darwin/src/MGLStyleLayer.mm

```diff
@@ -0,0 +1,38 @@
+#import "MGLStyleLayer.h"
+
+#import "MGLTypes.h"
+#import "MGLStyleLayer_Private.hpp"
+#include <mbgl/mbgl.hpp>
+
+#if TARGET_OS_IPHONE
+#import <UIKit/UIKit.h>
+#else
+#import <Cocoa/Cocoa.h>
+#endif
+
+@interface MGLStyleLayer()
+@property (nonatomic, weak) MGLMapView *mapView;
+@property (nonatomic) mbgl::Map *mbglMap;
+@property (nonatomic) mbgl::style::Layer *layer;
+@end
+
+@implementation MGLStyleLayer
+
+mbgl::Color MGLColorFromColor(MGLColor *color)
+{
+    if (!color)
+    {
+        return { 0, 0, 0, 0 };
+    }
+    CGFloat r, g, b, a;
+    [color getRed:&r green:&g blue:&b alpha:&a];
```

Comment:
This can crash on iOS 10 and macOS if the color belongs to a color space other than `NSCalibratedRGBColorSpace`. You’ll have to first convert to that color space using `-[MGLColor colorUsingColorSpaceName:]`. That conversion can fail in some cases, for instance in the case of a pattern color.